extensions/amp-3q-player/0.1/amp-3q-player.js

```diff
@@ -0,0 +1,190 @@
+      this.playerReadyResolver_ = resolve;
+    });
+
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    this.forwardEvents([VideoEvents.PLAY, VideoEvents.PAUSE], iframe);
+    this.applyFillContent(iframe, true);
+    this.element.appendChild(iframe);
+
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowfullscreen', 'true');
+
+    installVideoManagerForDoc(this.element);
+    videoManagerForDoc(this.element).register(this);
+  }
+
+  /** @override */
+  layoutCallback() {
+
+    const dataId = user().assert(
+        this.element.getAttribute('data-id'),
+        'The data-id attribute is required for <amp-3q-player> %s',
+        this.element);
+
+    const src = 'https://playout.3qsdn.com/' + dataId + '?autoplay=false&amp=true';
+    this.iframe_.src = src;
+
+    window.addEventListener('message',
+                            event => this.sdnBridge_(event));
+
+    return this.loadPromise(this.iframe_)
+      .then(() => {
+        this.element.dispatchCustomEvent(VideoEvents.LOAD);
+        this.playerReadyResolver_(this.iframe_);
+      });
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    return true;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  viewportCallback(visible) {
+    this.element.dispatchCustomEvent(VideoEvents.VISIBILITY, {visible});
+  }
+
+  /** @override */
+  pauseCallback() {
+    if (this.iframe_) {
+      this.pause();
+    }
+  }
+
+  /** @private */
+  sdnBridge_(event) {
+    const data = isObject(event.data) ? event.data : tryParseJson(event.data);
+    if (data === undefined) {
+      return;
+    }
+
+    switch (data.data) {
+      case 'playing':
+        this.element.dispatchCustomEvent(VideoEvents.PLAY);
+        break;
+      case 'pause':
+        this.element.dispatchCustomEvent(VideoEvents.PAUSE);
+        break;
+
+    }
+  }
+
+  // VideoInterface Implementation. See ../src/video-interface.VideoInterface
+
+  /** @override */
+  play(unusedIsAutoplay) {
+    this.playerReadyPromise_.then(() => {
+      this.iframe_.contentWindow./*OK*/postMessage('play2', '*');
+    });
+  }
+
+  /** @override */
+  pause() {
+    this.playerReadyPromise_.then(() => {
+      this.iframe_.contentWindow./*OK*/postMessage('pause', '*');
+    });
+  }
+
+  /** @override */
+  mute() {
+    this.playerReadyPromise_.then(() => {
+      this.iframe_.contentWindow./*OK*/postMessage('mute', '*');
+    });
+  }
+
+  /** @override */
+  unmute() {
+    this.playerReadyPromise_.then(() => {
+      this.iframe_.contentWindow./*OK*/postMessage('unmute', '*');
+    });
+  }
+
+  /** @override */
+  supportsPlatform() {
+    return true;
+  }
+
+  /** @override */
+  isInteractive() {
+    return true;
+  }
+
+  /** @override */
+  showControls() {
```

Comment:
is this not supported by 3q player? Considering `hideControls` is supported, it would be great to support this as well.